src/ranked_match.cc

```diff
@@ -196,6 +229,13 @@ bool RankedMatch::operator<(const RankedMatch& other) const
     if (diff != Flags::None)
         return (int)(m_flags & diff) > (int)(other.m_flags & diff);
 
+    if (m_path_component_matches.empty() ^ other.m_path_component_matches.empty())
+        return m_path_component_matches.size() > other.m_path_component_matches.size();
+    kak_assert(m_path_component_matches.size() == other.m_path_component_matches.size());
+    for (auto [lhs, rhs] : zip(m_path_component_matches, other.m_path_component_matches))
+        if (*lhs < *rhs)
+            return true;
```

Comment:
probably if we use the same recursive-matching for the basename (even if the query contains no slash) then we can make @raiguard's test work:

`kak_assert(preferred("luaremote", "src/script/LuaRemote.cpp", "tests/TestLuaRemote.cpp"));`

because `luaremote` is a prefix (or is it?) of the basename so the RankedMatch will win